### ISSUE TITLE
Downloading fails with Python 3.4, succeeds with 2.7

### ISSUE THREAD

**Issue by mikedld** [conan-io/conan#512]
I've seen #235 and #349 but my issue isn't connected to file size. I've created a package for SNMP++ and the [source tarball for version 3.3.5](http://www.agentpp.com/download/snmp++-3.3.5.tar.gz) there is just 657493 bytes in size. When installing and using Conan with Python 3.4.5, I'm getting this error:

```
SNMPXX/3.3.5@mikedld/stable: Installing package 1372ddd389cf0a35b656551e0c129859a506dac3
SNMPXX/3.3.5@mikedld/stable: Looking for package 1372ddd389cf0a35b656551e0c129859a506dac3 in remote 'mylocalremote' 
SNMPXX/3.3.5@mikedld/stable: WARN: Binary for 1372ddd389cf0a35b656551e0c129859a506dac3 not in remote: 
SNMPXX/3.3.5@mikedld/stable: Building your package in /home/mikedld/.conan/data/SNMPXX/3.3.5/mikedld/stable/build/1372ddd389cf0a35b656551e0c129859a506dac3
SNMPXX/3.3.5@mikedld/stable: Configuring sources in /home/mikedld/.conan/data/SNMPXX/3.3.5/mikedld/stable/source
[========================================          ]                  SNMPXX/3.3.5@mikedld/stable: WARN: Trying to remove dirty source folder
SNMPXX/3.3.5@mikedld/stable: WARN: This can take a while for big packages
ERROR: SNMPXX/3.3.5@mikedld/stable: Error in source() method, line 21
    download("http://www.agentpp.com/download/%s" % zip_name, zip_name, verify=False)
    Download failed, check server, possibly try again
slice indices must be integers or None or have an __index__ method
```

Using Python 2.7.12 the package builds just fine.

I'm using Conan 0.12.0 on Gentoo Linux x86_64 (installed with pip). Tell me if you need more information on this.

**Comment by memsharded** [comment]
Thanks for telling, no indeed it doesn't seem related to those issues. We are just in the process of changing all downloads and uploads to streaming from/to disk, but it is not related, as the ``tools.download()`` is separate functionality. Yes, seems a bug with Py3, will look, thanks!

**Comment by memsharded** [comment]
Issue fixed, but not released yet. Can run from develop branch from sources.
Will be released in conan 0.13.

**Comment by memsharded** [comment]
Released in conan 0.13.1

**Comment by levon-vardevanyan-solarwinds** [comment]
seems like this one is broken again somehow.
reproduced with 0.28.0 on ubuntu 17.10 and centos7.
installing 0.27.0 from source fixed the issue.
Let me know if you need more info.

`ERROR: SNMPXX/3.3.5@mikedld/stable: Error in source() method, line 21
	download("http://www.agentpp.com/download/%s" % zip_name, zip_name, verify=False)
	ConanConnectionError: Download failed, check server, possibly try again
Transfer interrupted before complete: 3266560 < 657493
`

**Comment by memsharded** [comment]
It is not exactly the same error, it seems. A minor release is open to fix this: https://github.com/conan-io/conan/tree/release/0.28.1

Can you please tell me which ``zip_name`` file are you trying to download, to check?
We might release it today or tomorrow, I will do an extra check with your info, thanks!

**Comment by memsharded** [comment]
I have tried with a couple of files from that domain, and not able to make it fail yet.

**Comment by levon-vardevanyan-solarwinds** [comment]
have u tried that topicstarter's tarball? even size in log is the same 657493.
now i'm trying to dive a little bit deeper into the package creation domain to understand the problem better and to be able to help u :)

**Comment by memsharded** [comment]
Link please? I am checking http://www.agentpp.com/download.html, but cannot see that topicstarter tarball. Thanks!

**Comment by levon-vardevanyan-solarwinds** [comment]
http://www.agentpp.com/download/snmp++-3.3.5.tar.gz and a link from the first message works for me.

**Comment by levon-vardevanyan-solarwinds** [comment]
Cool! That was fast :) Thanks.

**Comment by memsharded** [comment]
@levon-vardevanyan-solarwinds Released 0.28.1 fixing this, please try!

**Comment by levon-vardevanyan-solarwinds** [comment]
tried version installed by pip install conan = 0.28.1, looks great so far. seems like u fixed that, thank you.
